HW04.Rmd

```diff
@@ -1,126 +1,239 @@
-
-```{r stats?}
-
-
-```
-
-
-```{r police_2}
-
-```
-
-```{r stats_2?}
-
-
-```
-
-
-Finally, all together now:
-```{r police_3}
-
-```
-
-
-## Session info
-
-```{r}
-# always good to have this for reproducibility purposes
-devtools::session_info()
-```
-
+---
+title: "Exploring Gun Deaths in America"
+output: github_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
+```
+
+## Objectives
+* To explore gun death data curated by FiveThirtyEight [original data available here](https://github.com/fivethirtyeight/guns-data) by
+    * Month
+    * Intent
+    * Age and sex
+    * Education
+    * Season
+    * Intent by race
+    * Police-relatedness 
+
+## Read in the data
+
+The gun_deaths.csv is provided in the repository. 
+```{r load-data, message=FALSE, warning=FALSE}
+library(tidyverse)    # load tidyverse packages, including ggplot2
+library(viridis)      # color scheme
+library(scales)       # required for overlaying percentages over bar plots
+
+# set theme
+theme_set(theme_bw())
+
+my_colors <- c("#2a9d8f", "#E76F51", "#E9C46A", "#3a6b7e", "#F4A261")
+
+# read in the data
+gun_deaths <- read.csv('gun_deaths.csv', 
+                       na.strings=c("NotAvailable"), 
+                       colClasses = c("integer", "integer",  
+                                      "factor", "factor",  
+                                      "factor",  "integer",  
+                                      "factor", "integer",  
+                                      "factor",  "factor"))
+
+# I thought the following would work, and I can't figure out why it doesn't.
+# gun_deaths <- gun_deaths %>% mutate(race = replace(race, race == "Not Availabletive American/Not Availabletive Alaskan", "Native AmericaNative"))
+# It just replaces my string with NA, not the new string. :(
+
+# So I did this instead:
+gun_deaths <- gun_deaths %>% mutate(race = gsub("Not Availabletive American/Not Availabletive Alaskan", "Native American/Native Alaskan", race))
+gun_deaths <- gun_deaths %>% mutate(race = gsub("/", "/\n", race))
+
+```
+
+## Generating a data frame that summarizes the number of gun deaths per month and printing using `kable()`. 
+
+kable is part of the knitr package. Allows you to change column names and add captions and make pretty tables in your knitted document. Hint: set the argument format = "markdown"
+
+
+```{r month, warning=FALSE}
+deaths_per_month <- gun_deaths %>% count(month)
+
+knitr::kable(deaths_per_month, format = "markdown")
+```
+
+
+### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
+
+```{r month_chart}
+ggplot(deaths_per_month, aes(month.abb[month], n)) +
+  geom_bar(stat = "identity") +
+  xlab("Month") +
+  ylab("Gun Deaths") +
+  scale_x_discrete(limits = month.abb)
```

Comment:
I used the same function "scale_x_discrete" , but it was not working for me. I see that you set it to the month. abb and it looks great. I modified my code to reflect this as well (crediting you).